src/providers/UserProvider.tsx

```diff
@@ -23,16 +24,25 @@ export default function UserProvider({ children, data }: UserProviderProps): JSX
         if (!data.user.timeZone) {
           // Should validate bowser time zone with supported time-zones returned by backend
           const browserTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
-          const updateResponse = await updateUserProfile({ ...data.user, timeZone: browserTimeZone });
-          if (updateResponse.requestSucceeded) {
-            data.reloadUser();
-            snackbar.pageSuccess(
-              strings.formatString(
-                strings.UPDATED_TIMEZONE_MSG,
-                <Link to={APP_PATHS.MY_ACCOUNT}>{strings.MY_ACCOUNT}</Link>
-              ),
-              strings.formatString(strings.UPDATED_TIMEZONE_TITLE, browserTimeZone)
+          const backendTimeZonesResponse = await getTimeZones();
+          if (backendTimeZonesResponse.requestSucceeded) {
+            const foundTimeZone = backendTimeZonesResponse.timeZones?.find(
+              (timeZone) => timeZone.id === browserTimeZone
             );
+            const updateResponse = await updateUserProfile({
+              ...data.user,
+              timeZone: foundTimeZone ? browserTimeZone : strings.UTC,
```

Comment:
UTC will be a key in the map, we can use that value.

Reply:
actually this is fine, not sure what i was thinking.. the key is still 'UTC'

Reply:
actually no, this is not fine .. Mondays are hard..
The key will always be 'UTC', it will not be translated.
basically:
```
timeZone: foundTimeZone?.id || 'UTC'
```

Reply:
@constanzauanini i think i'm wrong again.. give me a minute let me get back.

Reply:
Ok, 
```
 {
      "id": "Etc/UTC",
      "longName": "Coordinated Universal Time"
    }
```
The code should be:
```
timeZone: foundTimeZone?.id || 'Etc/UTC',
```